src/sagemaker/jumpstart/types.py

```diff
@@ -1137,21 +1142,43 @@ def from_json(self, json_obj: Dict[str, Any]) -> None:
         if json_obj is None:
             return
 
-        self.preset_configs: str = {
+        self.preset_configs: Dict[str, JumpStartPresetConfig] = {
             alias: JumpStartPresetConfig(preset_config)
             for alias, preset_config in json_obj["preset_configs"].items()
         }
-        self.preset_config_rankings: str = json_obj["preset_config_rankings"]
+        self.preset_config_rankings: Optional[Dict[str, JumpStartPresetRanking]] = json_obj[
+            "preset_config_rankings"
+        ]
 
     def to_json(self) -> Dict[str, Any]:
         """Returns json representation of JumpStartInferencePresetConfig object."""
         json_obj = {att: getattr(self, att) for att in self.__slots__ if hasattr(self, att)}
         return json_obj
 
-    def get_resolved_config(self) -> JumpStartPresetConfig:
-        """Resolves the preset config."""
-        # TODO: return the best preset config based on config rankings
-        return self.preset_configs[0] if self.preset_configs else None
+    def get_top_config_from_ranking(
+        self,
+        ranking_name: str = "overall",
+        instance_type: Optional[str] = None,
+    ) -> JumpStartPresetConfig:
+        """Gets the best the preset config based on config ranking."""
+        if not self.preset_config_rankings or not self.preset_config_rankings.get(ranking_name):
+            return list(self.preset_configs.values())[0] if self.preset_configs else None
+
+        instance_type_attribute = (
+            "supported_inference_instance_types"
+            if self.scope == JumpStartScriptScope.INFERENCE
+            else "supported_training_instance_types"
+        )
+        rankings = self.preset_config_rankings.get(ranking_name)
+        for config_name in rankings.rankings:
+            resolved_config = self.preset_configs[config_name].resolved_config
+            if instance_type and instance_type not in getattr(
+                resolved_config, instance_type_attribute
+            ):
+                continue
+            return self.preset_configs[config_name]
+
+        return None
```

Comment:
will this lead to an exception being thrown later?

Reply:
No, it will not apply anything to the spec if there are no configs for this model. We should not fail for models without the alt config metadata.